File: src/main/java/com/randomlychosenbytes/jlocker/nonabstractreps/RestrictedUser.java
package com.randomlychosenbytes.jlocker.nonabstractreps;

import javax.crypto.SecretKey;

import static com.randomlychosenbytes.jlocker.manager.UtilsKt.encryptKeyWithString;

public class RestrictedUser extends User {

    public RestrictedUser(String password, SecretKey key) {
        super(password, encryptKeyWithString(key, password));
    }
}

File: src/main/java/com/randomlychosenbytes/jlocker/nonabstractreps/RestrictedUser.kt
package com.randomlychosenbytes.jlocker.nonabstractreps

import com.randomlychosenbytes.jlocker.manager.encryptKeyWithString
import javax.crypto.SecretKey

class RestrictedUser(password: String, key: SecretKey) : User(
    password, encryptKeyWithString(key, password)
)